eng/pipelines/doc-index.yml

```diff
@@ -0,0 +1,86 @@
+jobs:
+  - job: "Build"
+    variables:
+      - template: templates/variables/globals.yml
+    pool:
+      vmImage: "windows-2019"
+    steps:
+      - pwsh: |
+          Invoke-WebRequest -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
+          -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
+        workingDirectory: $(Build.BinariesDirectory)
+        displayName: Download and Extract DocFX
+
+      - task: CmdLine@2
+        displayName: Provision DocFX Directory
+        inputs:
+          script: $(Build.BinariesDirectory)/docfx/docfx.exe init -q
+          workingDirectory: $(Build.SourcesDirectory)
+          failOnStderr: true
+
+      - pwsh: New-Item -Path . -Name "templates" -ItemType "directory"
+        displayName: Create Template Directory
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/
+
+      # - task: CopyFiles@2
+      #   displayName: Copy Template to templates folder
+      #   inputs:
+      #     sourceFolder: $(Build.SourcesDirectory)/eng/tools/generate-static-index/static-files/templates
+      #     content: "**/*"
+      #     targetFolder: $(Build.SourcesDirectory)/docfx_project/templates
+      #     overWrite: true
+
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/templates/
+
+      - pwsh: |
+          Copy-Item "$(Build.SourcesDirectory)/eng/tools/generate-static-index/static-files/docfx.json" -Destination "$(Build.SourcesDirectory)/docfx_project/" -Force
+        displayName: Copy over docfx.json
+
+      - script: |
```

Comment:
Looks like you are doing a lot of one line script or pswh commands you should consider consolidating them into a smaller number of pwsh or script steps.

Reply:
So the way we have them right now is we have segregated the steps based on the logical purpose and also if there are any failures we can identify them easily